dotnet/samples/KernelSyntaxExamples/Example23_OpenAPIPlugin.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Plugins.OpenApi;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Examples;
+
+/// <summary>
+/// Examples to show how to create plugins from OpenAPI specs.
+/// </summary>
+public class Example23_OpenAPIPlugin : BaseTest
+{
+    /// <summary>
+    /// Example to show how to consume operation extensions and other metadata from an OpenAPI spec.
+    /// Try modifying the sample schema to simulate the other cases by
+    /// 1. Changing the value of x-openai-isConsequential to true and see how the function execution is skipped.
+    /// 2. Removing the x-openai-isConsequential property and see how the function execution is skipped.
+    /// </summary>
+    [Fact]
+    public async Task RunOpenAIPluginWithMetadataAsync()
+    {
+        Kernel kernel = new();
+
+        // This HTTP client is optional. SK will fallback to a default internal one if omitted.
+        using HttpClient httpClient = new();
+
+        // Create a sample OpenAPI schema that calls the github versions api, and has an operation extension property.
+        // The x-openai-isConsequential property is the operation extension property.
+        var schema =
+            "{" +
+            "  \"openapi\": \"3.0.1\"," +
+            "  \"info\": {" +
+            "    \"title\": \"Github Versions API\"," +
+            "    \"version\": \"1.0.0\"" +
+            "  }," +
+            "  \"servers\": [" +
+            "    {" +
+            "      \"url\": \"https://api.github.com\"" +
+            "    }" +
+            "  ]," +
+            "  \"paths\": {" +
+            "    \"/versions\": {" +
+            "      \"get\": {" +
+            "        \"x-openai-isConsequential\": false," +
```

Comment:
Make this true for the sake of the sample

Reply:
I did consider it, but version is not a consequential action, so I felt making it true would be misleading.  Instead I suggested in the xml comments that people should play around with the example by changing the value and seeing the outcome.